app/controllers/users/verify_account_controller.rb

```diff
@@ -25,10 +25,13 @@ def create
       throttled = Throttler::IsThrottledElseIncrement.call(
         current_user.id,
         :verify_gpo_key,
-        analytics: analytics,
       )
 
       if throttled
+        analytics.track_event(
```

Comment:
I think we need to add this to the index call as well. This case would only be hit if the user submitted the form while they're already throttled, but we don't show the form if they are.

Reply:
> I think we need to add this to the index call as well. This case would only be hit if the user submitted the form while they're already throttled, but we don't show the form if they are.

Ah, gotcha, that makes sense. I'll update.

On a related topic, @amathews-fs and I had discussed a bit if we could log the event only at the point they're initially being redirected to the throttle, vs. each and every time they're presented with a throttle page. I thought that might be what was happening here, as far as being the throttled submission. It might just be simpler and most consistent to always log the throttle on every view, and group by user / session if we need to aggregate.

Reply:
Unfortunately not sure it's possible. Current flow is for it to render index when it fails the result, and if that's the final attempt before throttling, `index` is the only place to reliably catch it.

And actually now that I look at it closer, it's weirdly doing a render of `:index` when the result is not successful, when it should probably be a redirect...

Reply:
> I think we need to add this to the index call as well. This case would only be hit if the user submitted the form while they're already throttled, but we don't show the form if they are.

Updated in 5dc89f7.

> And actually now that I look at it closer, it's weirdly doing a render of `:index` when the result is not successful, when it should probably be a redirect...

Sounds like a correct assessment to me. Dunno if it makes sense here, but I included it in dd85512. Flash errors would be shown in the redirect, yeah? I don't feel the tests have great coverage for those, despite the test case claiming "to show errors". 🤷

Reply:
it uses `flash.now[:error]` instead of `flash[:error]` to make the render show the flash message

Reply:
Test coverage was better than I expected, since it appeared to catch that. Should be better now in d129979, though I don't feel great about the chain of changes required with persisting errors that were previously read directly by the view from the prior form submission. 😬

Reply:
I _think_ it should be ok 🙂

Reply:
I tested it in the app. It seems to work fine, but only after I forcibly modify the code to trigger the `validate_pending_profile` validation, since from what I can tell, it's impossible for the user to actually trigger it, because its validation effectively occurs in a prior validation. We might want to clean this up (separately), though I guess it may be reasonable to have something explicitly checking the pending profile? 🤷 

1. https://github.com/18F/identity-idp/blob/376e0acda17f11512fec57646b811ff3d18ce004/app/forms/verify_account_form.rb#L5
2. https://github.com/18F/identity-idp/blob/376e0acda17f11512fec57646b811ff3d18ce004/app/forms/verify_account_form.rb#L49-L50
3. https://github.com/18F/identity-idp/blob/376e0acda17f11512fec57646b811ff3d18ce004/app/forms/verify_account_form.rb#L54-L55
4. https://github.com/18F/identity-idp/blob/376e0acda17f11512fec57646b811ff3d18ce004/app/forms/verify_account_form.rb#L33-L34